net/dns/resolver/forwarder.go

```diff
@@ -820,8 +820,8 @@ func (f *forwarder) sendTCP(ctx context.Context, fq *forwardQuery, rr resolverAn
 	return out, nil
 }
 
-// resolvers returns the resolvers to use for domain.
-func (f *forwarder) resolvers(domain dnsname.FQDN) []resolverAndDelay {
+// Resolvers returns the resolvers to use for domain.
+func (f *forwarder) Resolvers(domain dnsname.FQDN) []resolverAndDelay {
```

Comment:
exported thing returning unexported things is weird

Reply:
Yeah, we should either export `ResolverAndDelay`, or have another method that returns an exported type. Could you move `GetUpstreamResolvers` from `Resolver` to this type, and then have `Resolver.GetUpstreamResolvers` just call `r.forwarder.GetUpstreamResolvers`?

Reply:
Done